pronotepy/clients.py

```diff
@@ -510,28 +510,41 @@ def homework(
                 out.append(hw)
         return out
 
-    def get_recipients(self) -> List[dict]:
+    def get_recipients(self) -> List[dataClasses.Recipient]:
         """
-        Get all possible recipients for new discussion
+        Get recipients for new discussion
 
         Returns
         -------
-        List[dict]
+        List[dataClasses.Recipient]
         """
+        # add teacher
         data = {"onglet": {"N": 0, "G": 3}}
         recipients = self.post("ListeRessourcesPourCommunication", 131, data)[
             "donneesSec"
         ]["donnees"]["listeRessourcesPourCommunication"]["V"]
+        # add staff
         data = {"onglet": {"N": 0, "G": 34}}
         recipients += self.post("ListeRessourcesPourCommunication", 131, data)[
             "donneesSec"
         ]["donnees"]["listeRessourcesPourCommunication"]["V"]
 
-        return [r for r in recipients if r.get("avecDiscussion")]
+        return [dataClasses.Recipient(self, r) for r in recipients]
 
-    def new_discussion(self, subjet: str, message: str, recipient: List) -> None:
-        data = {"objet": subjet, "contenu": message, "listeDestinataires": recipient}
-        self.post("SaisieMessage", 131, data)
+    def new_discussion(
+        self, subjet: str, message: str, recipients: List[dataClasses.Recipient]
+    ) -> None:
+        """
+        Create a new discussion
+        """
+        recipients_json = [{"N": r.id, "G": r._type, "L": r.name} for r in recipients]
+        data = {
+            "objet": subjet,
+            "contenu": message,
+            "listeDestinataires": recipients_json,
+        }
+
+        print(self.post("SaisieMessage", 131, data))
```

Comment:
Oops looks like you left a debugging print here. Also don't you get back the discussion object from the API call? Would be nice to return it.

Reply:
No it return an empty dict

Reply:
Hmm, very pronote-like. Okay.